app/main.py

```diff
@@ -1,3 +1,55 @@
+from typing import Union, Optional
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union["Distance", int]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: int) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            return None
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            return None
+        return Distance(round(self.km / other, 2))
+
+    def __eq__(self, other: Union["Distance", int]) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __lt__(self, other: Union["Distance", int]) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other
+
+    def __gt__(self, other: Union["Distance", int]) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        return self.km > other
+
+    def __le__(self, other: Union["Distance", int]) -> bool:
+        return not self.__gt__(other)
+
+    def __ge__(self, other: Union["Distance", int]) -> bool:
+        return not self.__lt__(other)
```

Comment:
u can just write `<` or `>` here instead of __lt__, __gt__

Reply:
Done